nixos/modules/services/development/livebook.nix

```diff
@@ -0,0 +1,101 @@
+in
+{
+  options.services.livebook = {
+    enableUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to enable a user service for Livebook.  This implies
+        `installUserService`.
+      '';
+    };
+
+    installUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to install a user service for Livebook.  Unless
+        `enableUserService` is also set, users will have to manually activate
+        the service with `systemctl --user enable livebook`.
+      '';
+    };
+
+    environment = mkOption {
+      type = with types; attrsOf str;
+      description = lib.mdDoc ''
+        The environment to pass to the Livebook service.  You need to set at least
+        `LIVEBOOK_PASSWORD`.  See `livebook server --help` for other options.
+      '';
+      example = {
+        LIVEBOOK_PASSWORD = "a password at least 30 characters long";
+        LIVEBOOK_PORT = 30123;
+      };
+    };
+
+    erlang_node_short_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook";
+      description = "A short name for the distributed node.";
+    };
+
+    erlang_node_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook@127.0.0.1";
+      description = "The name for the app distributed node.";
+    };
+
+    port = mkOption {
+      type = with types; nullOr port;
+      default = null;
+      example = 30123;
+      description = "The port to start the web application on.";
+    };
+
+    cookie = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = "The cookie for the app distributed node.";
+    };
+
+    address = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = lib.mdDoc ''
+        The address to start the web application on.  Must be a valid IPv4 or
+        IPv6 address.
+      '';
+    };
+  };
+
+  config = mkIf (cfg.installUserService || cfg.enableUserService) {
+    systemd.user.services.livebook = {
+      environment = cfg.environment;
+      serviceConfig = {
+        Restart = "always";
+        ExecStart = ''\
+          ${pkgs.livebook}/bin/livebook server \
+          ${optionalString (cfg.erlang_node_short_name != null)
+            "--sname ${cfg.erlang_node_short_name}"} \
+          ${optionalString (cfg.erlang_node_name != null)
+            "--name ${cfg.erlang_node_name}"} \
+          ${optionalString (cfg.port != null)
+            "--port ${toString cfg.port}"} \
+          ${optionalString (cfg.cookie != null)
+            "--cookie ${cfg.cookie}"} \
+          ${optionalString (cfg.address != null)
+            "--ip ${cfg.address}"} \
+        '';
+      };
+      path = [ pkgs.bash ];
+    } // optionalAttrs cfg.enableUserService {
+      wantedBy = [ "default.target" ];
+    };
```

Comment:
It's not like the service is not wanted by anything if it's disabled, it just gets disabled as a side effect of not being wanted by anything. There is `systemd.user.services.<name>.enable` for this.

Reply:
Agreed, but see the comment on `installUserService` above.